avocado/utils/dmesg_utils.py

```diff
@@ -0,0 +1,114 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author: Praveen K Pandey <praveen@linux.vnet.ibm.com>
+#
+
+"""
+Module for manipulate dmesg while running test.
+"""
+
+
+import time
+import os
+from . import process, genio
+
+_dmesg_log_file = '%s/dmesglog-%s' % (genio._log_file_dir,
```

Comment:
Please don't import private properties of other module. Besides the way it's gathered in `genio` is wrong. The "right" way to get temp dir is `tempfile.gettempdir()`. Anyway even that would be dangerous as multiple processes or even users might attempt to load this module at the same time resulting in using the same file for their work, which might result in really odd behavior.

The biggest issue I have with this concept is that the file actually has no correlation with the system state. I mean the date/time is not the date/time the dmesg was gathered, it's just a date/time when the module was imported. I'd suggest simply relying on the user to provide a location and to remember it if necessary.